entity-service-impl/src/main/java/org/hypertrace/entity/data/service/EntityDataServiceImpl.java

```diff
@@ -108,6 +121,8 @@ public void upsert(Entity request, StreamObserver<Entity> responseObserver) {
           Entity.newBuilder(),
           entitiesCollection,
           responseObserver);
+      entityChangeEventGenerator.sendChangeNotification(
```

Comment:
in terms of error handling in general, we probably should not fail the call if this publish fails (that is, it should be outside the try).

Reply:
I think I did another round of review on the newer code before this set of comments :)